ShoppingComponents/shoppingCart.py

```diff
@@ -0,0 +1,84 @@
+class Cart():
+    def __init__(self):
+        self.items = {}
+
+    '''
+	Add item to shopping cart 
+	'''
+
+    def addToCart(self, item):
+        if item.itemName in self.items:
+            new_quantity = item.getQuantity()
+            old_quantity = self.items[item.itemName].getQuantity()
+            self.items[item.itemName].setQuantity(new_quantity + old_quantity)
+        else:
+            name = item.itemName
+            self.items[name] = item
+        return
+
+    '''
+    Used to remove items in shopping cart with numbers 
+    '''
+
+    def editCart(self, item, amount):
+        if item.itemName not in self.items:
+            print (item.itemName + " not in shopping cart! ")
+            return False
+
+        else:
+            number = self.items[item.itemName].getQuantity()
+            if number <= amount:
+                self.items.pop(item.itemName)
+            else:
+                self.items[item.itemName].setQuantity(number - amount)
+
+        return True
+
+    '''
+	remove item from shopping cart, item is a string 
+	'''
+
+    def removeItem(self, item):
+        if item not in self.items:
+            print item + " not in shopping cart"
+            return False
+        self.items.pop(item)
+        return True
+
+    '''
+	Print every item + quantity in the shopping cart. 
+	'''
+
+    def toString(self):
+        lst = []
+        for (k, item) in self.items.iteritems():
+            lst.append((k, item.getQuantity()))
+        print (lst)
+        return
+
+    '''
+	list everything in the shopping cart 
+	'''
+
+    def printCart(self):
+        print ("name".ljust(40) + "quantity".ljust(20) + "price".ljust(20))
+        print ("-" * 65)
+        for (k, item) in self.items.iteritems():
+            name = k
+            quantity = item.getQuantity()
+            price = item.getPrice()
+            totalP = float(quantity) * price
+            print (name.ljust(40) + str(quantity).ljust(20) + str(totalP).ljust(20))
+        return
```

Comment:
I don't really have a problem with this, other than the lack of consistency with decimal places (i.e. 43.8 instead of 43.80). For the sake of demonstration, though, we can argue that this method has too many concerns. Here is an example that breaks the concerns into smaller ones, and achieves a more DRY (don't repeat yourself) approach. It reduces the likelihood of misalignment, and shifts to a more functional approach:

<img width="472" alt="screenshot 2017-12-16 15 06 56" src="https://user-images.githubusercontent.com/933621/34073918-cf8073e4-e272-11e7-8b0b-f6e8f9c23653.png">


```python
    def printCart(self):
        self.printHeader();
        total = 0

        for (k, item) in self.items.iteritems():
            name = k.strip()
            quantity = item.getQuantity()
            price = item.getPrice()
            totalP = float(quantity) * price
            total += totalP

            self.printItem(name, quantity, totalP);

        self.printTotal(total)
        return

    def printHeader(self):
        print("")
        self.printRow("Name", "Quantity", "Price")
        print ("-" * 65)

    def printItem(self, name, quantity, price):
        self.printRow(name, str(quantity), self.formatPrice(price))

    def printTotal(self, total):
        print ("=" * 65)
        self.printRow("Total", "", self.formatPrice(total))
        print("")

    def printRow(self, name, quantity, price):
        print (name.ljust(40) + quantity.ljust(20) + price.ljust(20))

    def formatPrice(self, price):
        return "{0:.2f}".format(price)
```